app/script/view_model/FaviconViewModel.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.viewModel = z.viewModel || {};
+
+z.viewModel.FaviconViewModel = class FaviconViewModel {
+  constructor(dispatcher) {
+    this.dispatcher = dispatcher;
+    this.unreadCount = ko.observable(0);
+
+    this.unreadCountSubscription = this.unreadCount.subscribe(this._updateFavicon.bind(this));
+    this.dispatcher.subscribe(z.event.WebApp.LIFECYCLE.UNREAD_COUNT, this, this._updateUnreadCount);
+  }
+
+  _updateUnreadCount(unreadCount) {
+    this.unreadCount(unreadCount);
+  }
+
+  _updateFavicon(unreadCount) {
+    const link = document.querySelector("link[rel*='shortcut icon']") || document.createElement('link');
+    link.type = 'image/x-icon';
+    link.rel = 'shortcut icon';
+    link.href = unreadCount != 0 ? '/image/favicon-badge.ico' : '/image/favicon.ico';
```

Comment:
Let's not forget about the PNG favicons, too.

Reply:
Turns out we don't need to update the png files as they are used for other purposes (like displaying large icons for the website you often browse on your browser).